### ISSUE TITLE
Consider remove `DatabaseEngine`

### ISSUE THREAD

**Issue by drmingdrmer** [datafuselabs/databend#2205]
looks good to me. 

I've not thought about it thoroughly, intuitionally,  there **might** be a .. "handful" of tasks there to remove it from `Database`.

If we could drop the concept of `database engine`,  things might be easier (for this case only).  for reference only @drmingdrmer

_Originally posted by @dantengsky in https://github.com/datafuselabs/databend/issues/2180#issuecomment-940603140_

**Comment by sundy-li** [comment]
What if we have proxy datasource bridges like `DatabaseMySQL`, `DatabasePg`

**Comment by dantengsky** [comment]
> What if we have proxy datasource bridges like `DatabaseMySQL`, `DatabasePg`

yeah, that's a hard tradeoff. 

once @BohuTANG suggested that if the database engine concepts do block our progress, it might be acceptable that, we could only provide `proxy datasource bridges` at table level. i.e  supports `create table xxx engine= PG conn_url=xxxx ....`, but not `create database engine=PG para=xxxx ...`

It is too hard to balance for me, fortunately not an urgent req, I just left it as it was.

**Comment by BohuTANG** [comment]
@dantengsky
You are right, the engine can be bind with table(MySQL does) not database(ClickHouse).

**Comment by BohuTANG** [comment]
> > What if we have proxy datasource bridges like `DatabaseMySQL`, `DatabasePg`
> 
> yeah, that's a hard tradeoff.
> 
> once @BohuTANG suggested that if the database engine concepts do block our progress, it might be acceptable that, we could only provide `proxy datasource bridges` at table level. i.e supports `create table xxx engine= PG conn_url=xxxx ....`, but not `create database engine=PG para=xxxx ...`
> 
> It is too hard to balance for me, fortunately not an urgent req, I just left it as it was.

We can remove the database engine, or there is really too much to consider.

**Comment by drmingdrmer** [comment]
It's about time to close it!
